internal/zero/analytics/sessions.go

```diff
@@ -0,0 +1,52 @@
+package analytics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pomerium/pomerium/pkg/grpc/databroker"
+	"github.com/pomerium/pomerium/pkg/grpc/session"
+	"github.com/pomerium/pomerium/pkg/protoutil"
+)
+
+var (
+	sessionTypeURL = protoutil.GetTypeURL(new(session.Session))
+)
+
+// CurrentUsers returns a list of users active within the current UTC day
+func CurrentUsers(
+	ctx context.Context,
+	client databroker.DataBrokerServiceClient,
+) ([]string, error) {
```

Comment:
I'm wondering if it would be worthwhile to avoid keeping all user IDs in memory while updating the counters? You could imagine an alternative design where this method accepts some callback that is invoked per user ID. Although I don't have a clear sense of what user/session volume this needs to support.

Actually, doesn't the identity manager keep all current user sessions in memory already? Do you think it would be any simpler to integrate analytics reporting with the identity manager code rather than adding a totally separate syncer here?

Reply:
> Do you think it would be any simpler to integrate analytics reporting with the identity manager code

there might be benefits for doing that. if we move it there, it would be running for any kind of deployment and not just zero. let me look into that. 

the zero control loop can then run counter syncer and report the changes to the Zero. the cloud interaction would continue being severed from rest of the core.

Reply:
if we move it to identity_manager, this is where it can be hooked up to: 
https://github.com/pomerium/pomerium/blob/3b2bdd059a02b20654d19e2e9b04a3359d0e28cc/internal/identity/manager/manager.go#L389

the counter logic would have to maintain an additional control loop to periodically store the counters.

Reply:
after some prototyping I would prefer to leave it here for now.